qiskit_nature/drivers/second_quantization/electronic_structure_molecule_driver.py

```diff
@@ -38,34 +39,43 @@ class ElectronicStructureDriverType(Enum):
     @staticmethod
     def driver_class_from_type(
         driver_type: "ElectronicStructureDriverType",
+        method: MethodType,
     ) -> Type[ElectronicStructureDriver]:
         """
         Get driver class from driver type
 
         Args:
             driver_type:type of driver to be used. If `AUTO` is selected, it will use
-                        the first driver installed in the following order:
+                        the first driver installed and that supports the given method
+                        in the following order:
                         `PYSCF`, `PSI4`, `PYQUANTE`, `GAUSSIAN`
+            method: Used to verify if the driver supports it.
 
         Returns:
             driver class
 
         Raises:
             MissingOptionalLibraryError: Driver not installed.
+            NotSupportedMethodError: method not supported by driver.
         """
         driver_class = None
         if driver_type == ElectronicStructureDriverType.AUTO:
-            missing_error = None
+            error = None
             for item in ElectronicStructureDriverType:
                 if item != ElectronicStructureDriverType.AUTO:
                     try:
-                        driver_class = ElectronicStructureDriverType.driver_class_from_type(item)
+                        driver_class = ElectronicStructureDriverType.driver_class_from_type(
+                            item, method
+                        )
                         break
                     except MissingOptionalLibraryError as ex:
-                        if missing_error is None:
-                            missing_error = ex
+                        if error is None:
+                            error = ex
+                    except NotSupportedMethodError as ex:
```

Comment:
You could also put these on one line if you wanted to:
```python
except (MissingOptionalLibraryError, NotSupportedMethodError) as ex:
```

Reply:
Changed it, thanks.